packages/web3-providers-ws/test/integration/web_socket_provider_integration.test.ts

```diff
@@ -0,0 +1,338 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* eslint-disable jest/no-done-callback */
+import {
+	EthExecutionAPI,
+	JsonRpcId,
+	Web3APIPayload,
+	DeferredPromise,
+	JsonRpcResponse,
+	JsonRpcNotification,
+	JsonRpcSubscriptionResult,
+} from 'web3-common';
+import express from 'express';
+import { createProxyMiddleware } from 'http-proxy-middleware';
+
+import { Server } from 'http';
+import { Web3WSProviderError } from 'web3-errors';
+import WebSocketProvider from '../../src/index';
+// import { Web3WSProviderError } from '../../src/errors';
+import { WSRequestItem, OnCloseEvent } from '../../src/types';
+
+import {
+	getSystemTestProvider,
+	describeIf,
+	getSystemTestAccounts,
+} from '../fixtures/system_test_utils';
+
+describeIf(getSystemTestProvider().includes('ws'))(
+	'WebSocketProvider - implemented methods',
+	() => {
+		let clientWsUrl: string;
+		let accounts: string[];
+		let webSocketProvider: WebSocketProvider;
+		let jsonRpcPayload: Web3APIPayload<EthExecutionAPI, 'eth_getBalance'>;
+		// helper function
+		let currentAttempt = 0;
+		const waitForOpenConnection = async (provider: WebSocketProvider, status = 'connected') => {
+			return new Promise<void>((resolve, reject) => {
+				const maxNumberOfAttempts = 10;
+				const intervalTime = 5000; // ms
```

Comment:
This seems unnecessary large waiting time for reconnect.

Reply:
I took the number from here:

https://github.com/ChainSafe/web3.js/blob/07fa78a1922d362186a87cea3c2d5b3405229e53/packages/web3-providers-ws/src/index.ts#L85

I'll reduce it to 1000